### PR TITLE
draft for steerlm 2.0

### PR DESCRIPTION
**Pull request by yidong72**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo_aligner/models/nlp/gpt/gpt_steerlm_model.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                    "input_ids": batch["tokens"],
+                    "position_ids": batch["position_ids"],
+                    "attention_mask": None if self.get_attention_mask_from_fusion else batch["attention_mask"],
+                    "labels": batch["labels"],
+                }
+
+                output_tensor = model(**forward_args)
+
+                def id_func(output_tensor):
+                    nll = self.loss_func(batch["loss_mask"], output_tensor)
```
Comment (gshennvm) [inline nemo_aligner/models/nlp/gpt/gpt_steerlm_model.py:150]: do you not need to compute the log probs here? at the end of the model only thing we have are the logits. it needs to be normalized

**Review by gshennvm — CHANGES_REQUESTED** [review]
